opentrons-ai-server/api/handler/fast.py

```diff
@@ -227,6 +294,22 @@ async def create_chat_completion(
         ) from e
 
 
+def _determine_protocol_action(body: ChatRequest) -> str:
+    """Determine protocol action based on message content."""
+    # Default action is update
+    protocol_action = "update"
+
+    # Check if first history message indicates a create action
+    if body.history:
+        message = body.history[0]
+        if isinstance(message, dict) and "content" in message and message["content"] is not None:
```

Comment:
this can be cleaned up a little by doing
```suggestion
        if isinstance(message, dict) and message.get("content") is not None:
```

Reply:
got it thanks!